src/services/codefixes/fixAddMissingMember.ts

```diff
@@ -80,37 +80,38 @@ namespace ts.codefix {
         },
     });
 
-    function getAllSuperClasses(cls: ClassLikeDeclaration | undefined, checker: TypeChecker): ReadonlyArray<ClassLikeDeclaration> {
+    function getAllSupers(decl: ClassOrInterface | undefined, checker: TypeChecker): ReadonlyArray<ClassOrInterface> {
         const res: ClassLikeDeclaration[] = [];
-        while (cls) {
-            const superElement = getClassExtendsHeritageElement(cls);
+        while (decl) {
+            const superElement = getClassExtendsHeritageElement(decl);
             const superSymbol = superElement && checker.getSymbolAtLocation(superElement.expression);
             const superDecl = superSymbol && find(superSymbol.declarations, isClassLike);
             if (superDecl) { res.push(superDecl); }
-            cls = superDecl;
+            decl = superDecl;
         }
         return res;
     }
 
+    type ClassOrInterface = ClassLikeDeclaration | InterfaceDeclaration;
     interface InfoBase {

```

Comment:
I'm not sure I understand the way the Info types are broken down.  Why isn't there a separate `InfoKind` for interface?  Is that uninteresting?  What's the advantage of having a base type if the members are redeclared in the subtypes?  Is there an advantage/disadvantage compared to just unioning the two subtypes?  Why does `EnumInfo` have fewer fields?

Reply:
Oh, I see that `Info` _is_ declared as a union type.  Why does `InfoBase` exist at all then?

Reply:
Having fewer cases in a union is better since each union type needs its own code to handle. `EnumInfo` has fewer fields because the code handling it is a completely separate path that does different things from the `ClassOrInterfaceInfo` case. That's necessary because enums work much differently than classes/interfaces, but for class/interface the handling is mostly the same and it's better to just test `isInterfaceDeclaration` when necessary.